libs/Microsoft.MixedReality.WebRTC/PeerConnection.cs

```diff
@@ -280,11 +280,201 @@ public struct VideoCaptureFormat
+
+        /// <summary>
+        /// List of stream IDs to associate the transceiver with.
+        /// </summary>
+        public List<string> StreamIDs;
+    }
+
+    /// <summary>
+    /// Wrapper for an event possibly delayed.
+    /// </summary>
+    public class DelayedEvent
+    {
+        /// <summary>
+        /// The event handler.
+        /// </summary>
+        public Action Event;
+
+        /// <summary>
+        /// Begin suspending <see cref="Event"/>. This must be matched with a call
+        /// to <see cref="EndSuspend"/>. During this time, calling <see cref="Invoke"/>
+        /// does not invoke the event but instead queue it for later invoking by
+        /// <see cref="EndSuspend"/>.
+        /// </summary>
+        public void BeginSuspend()
+        {
+            lock (_lock)
+            {
+                ++_suspendCount;
+            }
+        }
+
+        /// <summary>
+        /// End suspending <see cref="Event"/> and invoke it if any call to <see cref="Invoke"/>
+        /// was made since the first <see cref="BeginSuspend"/> call.
+        /// </summary>
+        public void EndSuspend()
+        {
+            Action cb = null;
+            lock (_lock)
+            {
+                Debug.Assert(_suspendCount > 0);
+                --_suspendCount;
+                if (_eventPending && (_suspendCount == 0))
+                {
+                    cb = Event;
+                    _eventPending = false;
+                }
+            }
+            cb?.Invoke();
+        }
+
+        /// <summary>
+        /// Try to invoke <see cref="Event"/>, either immediately if not suspended, or later
+        /// when the last <see cref="EndSuspend"/> call stops suspending it.
+        /// </summary>
+        /// <param name="async">If the event is not suspended, and therefore is invoked immediately,
+        /// then invoke it asynchronously from a worker thread.</param>
+        public void Invoke(bool async = false)
+        {
+            Action cb = null;
+            lock (_lock)
+            {
+                Debug.Assert(_suspendCount >= 0);
+                if (_suspendCount > 0)
+                {
+                    _eventPending = true;
+                }
+                else
+                {
+                    cb = Event;
+                }
+            }
+            if (cb != null)
+            {
+                if (async)
+                {
+                    Task.Run(() => cb.Invoke());
+                }
+                else
+                {
+                    cb.Invoke();
+                }
+            }
+        }
+
+        /// <summary>
+        /// Try to invoke <see cref="Event"/> asynchronously.
+        /// This is equivalent to <code>Invoke(async: true)</code>.
+        /// </summary>
+        public void InvokeAsync() => Invoke(async: true);
+
+        /// <summary>
+        /// Lock for renegotiation event suspending variables:
```

Comment:
nit: this mentions renegotiation but the class is generic.

Reply:
Legacy comment from the time it was not in its own class, will fix.

Reply:
Fixed.